### ISSUE TITLE
Better error message

### ISSUE THREAD

**Issue by nirbhayc** [codership/galera#369]
The following error message does not seem to be very apt at explaining the underlying problem.

```
2015-11-22 11:57:06 140044815820544 [Note] WSREP: Member 0.0 (my_node2) requested state transfer from '*any*'. Selected 1.0 (my_node1)(SYNCED) as donor.
2015-11-22 11:57:06 140044815820544 [Note] WSREP: Shifting PRIMARY -> JOINER (TO: 0)
2015-11-22 11:57:06 140045096625920 [Note] WSREP: Requesting state transfer: success, donor: 1
2015-11-22 11:57:06 140044815820544 [Warning] WSREP: 1.0 (my_node1): State transfer to 0.0 (my_node2) failed: -2 (No such file or directory)
2015-11-22 11:57:06 140044815820544 [ERROR] WSREP: gcs/src/gcs_group.cpp:int gcs_group_handle_join_msg(gcs_group_t*, const gcs_recv_msg_t*)():731: Will never receive state. Need to abort.
```

For instance, in "WSREP: 1.0 (my_node1): State transfer to 0.0 (my_node2) failed: -2 (No such file or directory)" :
- What's "1.0" ? (node index : 1, node segment : 0)
- What's "0.0"?  (node index : 0, node segment : 0)
- Which 'file or directory' is being referred?

Perhaps an additional hint to user to refer to donor's error log?

Thanks for Elena for reporting this.

**Comment by nirbhayc** [comment]
Also, if wsrep_node_name is not set :
WSREP: 1.0 (wheezy-64): State transfer to 0.0 (wheezy-64) failed: -2 (No such file or directory)

So, it wsrep_node_name is not set, one idea would be to generate one
https://github.com/docker/docker/blob/master/pkg/namesgenerator/names-generator.go

**Comment by ayurchen** [comment]
I would agree about referral to donor's error log. If wsrep_node_name is not set it defaults to the hostname. Inventing arbitrary names does not sound to help to diagnose the problem - how do we know which node is which?

**Comment by ayurchen** [comment]
And even referring to donor's error log is not good. This is a generic library. It cant know about error logs.

**Comment by temeo** [comment]
Maybe the message should refer to donor log (without error) ;). Donor must have some kind of log, doesn't it?

**Comment by ayurchen** [comment]
> Donor must have some kind of log, doesn't it?

who knows? So would changing

```
"WSREP: 1.0 (my_node1): State transfer to 0.0 (my_node2) failed: -2 (No such file or directory)"
```

to

```
"WSREP: 1.0 (my_node1): State transfer to 0.0 (my_node2) failed on donor (code 2)"
```

make it clearer?

**Comment by temeo** [comment]
Yes.
